modules/spring-config/src/main/java/org/mule/config/spring/OptionalObjectsController.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.config.spring;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Keeps track of keys that have been marked as optional
+ * on a registry-bootstrap file. It also tracks the optional keys
+ * that couldn't in fact be instantiated and are discarded
+ *
+ * @since 3.7.0
+ */
+public class OptionalObjectsController
```

Comment:
Dont understand the idea. Optional objects are still in the registry? If that is true another bean could reference an optional bean that is going to be remove before intialization, which would be an error